### PR TITLE
Only update core store if data has changed

### PR DESCRIPTION
**Pull request by derrickmehaffy**
### What does it do?

Modifies the coreStore set method to only update data if nothing has changed, else create new

### Why is it needed?

Suggestion from a customer patch they are using to prevent writes that aren't needed during startup

### How to test it?

TBD

### Related issue(s)/PR(s)

Internal TID4799

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
Building experimental for testing: https://github.com/strapi/strapi/actions/runs/6424660625/job/17445751519

**Comment by derrickmehaffy** [comment]
Looks like main is broken right now, waiting for engineers to fix and will build a new experimental after.

**Comment by derrickmehaffy** [comment]
Lint failures aren't me so not sure if we have main borked atm :/

File: packages/core/strapi/src/services/core-store.ts [hunk 1]
```diff
@@ -126,13 +126,36 @@ const createCoreStore = ({ db }: { db: Database }) => {
...
+        return db.query('strapi::core-store').update({
+          where: { id: data.id },
+          data: {
+            value: newVal,
+            type: typeof value,
+          },
+        });
+      } else {
+        return {};
+      }
```
Comment (joshuaellis) [inline packages/core/strapi/src/services/core-store.ts:158]: I think this is your lint fail

**Comment by derrickmehaffy** [comment]
Closing as this would most likely be a breaking change within v4 and could have unintended side effects with some community plugins.

**Pull request closed without merging**